src/dbinterface.jl

```diff
@@ -384,9 +384,16 @@ Base.IteratorEltype(::Type{Cursor{false, knownlength}}) where {knownlength} = Ba
 Base.eltype(x::Cursor{false, knownlength}) where {knownlength} = Row
 
 function Base.iterate(x::Cursor{false}, st=1)
-    status = API.SQLFetchScroll(API.getptr(x.stmt), API.SQL_FETCH_NEXT, 0)
-    while status == API.SQL_STILL_EXECUTING
+    while true
         status = API.SQLFetchScroll(API.getptr(x.stmt), API.SQL_FETCH_NEXT, 0)
+        if status == API.SQL_SUCCESS_WITH_INFO
+            @warn(API.diagnostics(x.stmt))
+            break
+        end
+        if status == API.SQL_ERROR || status == API.SQL_INVALID_HANDLE
+            error(API.diagnostics(x.stmt))
+        end
+        status == API.SQL_STILL_EXECUTING || break
     end
     status == API.SQL_SUCCESS || status == API.SQL_SUCCESS_WITH_INFO || return nothing
```

Comment:
```suggestion
```
Hmmm, I think we can remove this now? Since we're already erroring if it's an error?

Reply:
I think we can't, because it gets triggered after the last row is fetched. We need it to return nothing as requested by Base.iterate API [here](https://github.com/JuliaLang/julia/blob/36034abf26062acad4af9dcec7c4fc53b260dbb4/base/essentials.jl#L897-L903)

Reply:
that code is `SQL_NO_DATA`; I've updated the PR to take recognize its existence implicitly 😄